core/.js/src/main/scala/io/udash/utils/FileService.scala

```diff
@@ -0,0 +1,45 @@
+package io.udash.utils
+
+import java.io.IOException
+
+import org.scalajs.dom._
+import scalatags.JsDom.all._
+
+import scala.scalajs.js
+import js.JSConverters._
+import js.typedarray._
+
+import scala.concurrent._
+import scala.util.Try
+
+object FileService {
+  def save(filename: String, bytes: Array[Byte], mimeType: String = "application/octet-stream"): Unit = {
+    val jsBytes = Array[js.Any](bytes.toTypedArray).toJSArray
```

Comment:
You can avoid the copy by creating a Javascript array directly: `js.Array[js.Any](bytes.toTypedArray)`